### PR TITLE
better support for -1,-2,-3 command line parameters

### LINKED ISSUE #1643: the `-1` parameter does not work
The `-1` parameter was introduced to start the game in a window with original 200p height. However, we force the minimal window size to be at least twice that size, so the window gets resized immediately. Now, the `-1` parameter works exactly as the `-2` parameter. This could be mitigated by the following patch:

```patch
--- a/src/i_video.c
+++ b/src/i_video.c
@@ -1656,8 +1656,8 @@ static void CreateSurfaces(int w, int h)
 
     I_InitDiskFlash();
 
-    SDL_SetWindowMinimumSize(screen, video.unscaledw * 2,
-                             use_aspect ? ACTUALHEIGHT * 2 : SCREENHEIGHT * 2);
+    SDL_SetWindowMinimumSize(screen, video.unscaledw,
+                             use_aspect ? ACTUALHEIGHT : SCREENHEIGHT);
 
     if (!fullscreen)
     {
@@ -1701,8 +1701,8 @@ void I_ResetScreen(void)
     CreateSurfaces(video.pitch, video.height);
     ResetLogicalSize();
 
-    SDL_SetWindowMinimumSize(screen, video.unscaledw * 2,
-                             use_aspect ? ACTUALHEIGHT * 2 : SCREENHEIGHT * 2);
+    SDL_SetWindowMinimumSize(screen, video.unscaledw,
+                             use_aspect ? ACTUALHEIGHT : SCREENHEIGHT);
 }
 
 void I_ShutdownGraphics(void)
```

So why was this introduced in the first place? Back in the days when we only had a choice of two distinct resolutions ("low" and "high"), we needed to make sure that the windows size would not get smaller than the actual frame buffer resolution to prevent the content from being scaled down. However, now we have more than these two resolutions and we could extend the meaning of the `-1`, `-2`, `-3` etc. parameters to not only set the window size, but also the frame buffer size while at it. This could be achieved with the following patch (or am I missing something?):

```patch
diff --git a/src/i_video.c b/src/i_video.c
index 88afbc0d..9af817aa 100644
--- a/src/i_video.c
+++ b/src/i_video.c
@@ -330,6 +330,9 @@ static void AdjustWindowSize(void)
     {
         window_width = scalefactor * video.unscaledw;
         window_height = scalefactor * ACTUALHEIGHT;
+        current_video_height = scalefactor * SCREENHEIGHT;
+        dynamic_resolution = (scalefactor > 1);
+        scalefactor = 0;
     }
     else if (old_w > 0 && old_h > 0)
     {
```

I often use the `-1` parameter during development when I want to make sure the port still compiles and runs as expected, but I don't want it to necessarily get my full attention by running full screen or in a full sized window.

### PR DESCRIPTION
**Pull request by rfomin**
Fix #1643

### CODE REVIEW

File: src/i_video.c [hunk 1]
```diff
@@ -1656,8 +1671,8 @@ static void CreateSurfaces(int w, int h)
 
     I_InitDiskFlash();
 
-    SDL_SetWindowMinimumSize(screen, video.unscaledw * 2,
-                             use_aspect ? ACTUALHEIGHT * 2 : SCREENHEIGHT * 2);
+    SDL_SetWindowMinimumSize(screen, video.unscaledw,
+                             use_aspect ? ACTUALHEIGHT : SCREENHEIGHT);
```
Comment (rfomin) [inline src/i_video.c:1675]: Should we leave the minimum size at 400p for `scalefactor == 0`? I don't like it when I accidentally resize the window and make it too small.

Comment (fabiangreffrath) [reply src/i_video.c:1675]: Yes, good idea.

Comment (rfomin) [reply src/i_video.c]: Done in https://github.com/fabiangreffrath/woof/pull/1644/commits/7201533f13714ce4e1ab2c52a41081d4ab051275

**Review by fabiangreffrath — APPROVED** [review]
Seems to work fine, thank you!

**Pull request merged**